packages/admin/cms-admin/src/userPermissions/UserGrid.tsx

```diff
@@ -48,15 +54,49 @@ export const UserGrid = () => {
             sortable: false,
             pinnable: false,
             filterable: false,
-            renderCell: (params) => (
-                <IconButton
-                    onClick={() => {
-                        stackApi.activatePage("edit", params.id.toString());
-                    }}
-                >
-                    <Edit color="primary" />
-                </IconButton>
-            ),
+            renderCell: (params) => {
+                const isCurrentUser = params.row.id === currentUser.id;
+                const isImpersonated = currentUser.impersonated;
+                return (
+                    <>
+                        {isAllowed("impersonation") && (
```

Comment:
Here would be better for the readability to use a own Component and use the early return pattern for the condition you use.